portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java

```diff
@@ -0,0 +1,53 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.sps.data.Comment;
+import com.google.gson.Gson;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.FetchOptions.Builder;
+import java.util.ArrayList;
+import java.util.List;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that deletes comments data */
+@WebServlet("/delete-data")
+public class DeleteDataServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    Query query = new Query("Comment");
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    for (Entity taskEntity : datastore.prepare(query).asIterable()) {
+      datastore.delete(taskEntity.getKey());
+    }
+
+    // Redirect back to the HTML page. idk if this is right or if I'm meant to send an empty response.
+    response.sendRedirect("/index.html#comments");
```

Comment:
You're modifying the data, so I think this is the right thing to do. Moreso out of my own lack of knowledge and curiosity, but what happens if you do send an empty response?
I think the problem should be looked at as one of: "After I modify the data, how do I want to present that change to the user? i.e. you're controlling the user flow" 
Not the best example, but maybe after you delete all the comments you think a user doesn't want to see the comment section so you take them to another part of the page/a different page.

Reply:
Yeah the behavior looks fine to me. I'm not sure what happens when sending an empty response; I didn't try it. In the walkthrough they said _"Your servlet can just return an empty response when it is done."_ But I'm not sure how exactly to do that - does that mean something like this? `response.getWriter().println("");` Or if there is a specific way to send an empty response? I tried looking it up but couldn't find anything.

Reply:
Okay, I just commented out the line entirely, sending no response at all, and it seems to be doing the exact same thing. I'm kind of confused on how that's working exactly, but I think I'm just going to leave the line in there since that makes sense to me.

Reply:
I think the important part to remember here is your deleteComments() JS function you defined. The flow that is occurring is:
User clicks on "Delete All Comments" button -> POST request to "/delete-data" is made -> doPost from DeleteDataServlet deletes the data from the database -> Return a response -> Make a call to getComments() once the fetch Promise returns -> getComments() makes a GET request and returns an empty response with no comments -> UI updated to have no comments
 
The response from the POST request to /delete-data doesn't matter in this case because you make the follow-up request to getComments and are essentially ignoring the response in the "then" of the promise:
"
.then(
    getComments()
  );
"

Reply:
Oh, I see! Thank you, that makes sense.